server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -557,6 +560,23 @@ protected long buildTookInMillis(long startTimeNanos) {
         return TimeUnit.NANOSECONDS.toMillis(relativeTime() - startTimeNanos);
     }
 
+    private static final class ReducedRequestInfo {
+        // todo: can stream mapping be parallel on multi threads? If so, this needs to be volatile
```

Comment:
Our ForbiddenAPIs disallow doing parallel stream processing, so I would not make these volatile for now.